### ISSUE TITLE
Tabbing Gets Stuck in Code Editor

### ISSUE THREAD

**Issue by Jatin24062005** [processing/p5.js-web-editor#3437]
### What is your operating system?

None

### Actual Behavior

When using the [p5.js Web Editor](https://editor.p5js.org/), pressing the Tab key inside the Code Editor causes the focus to get trapped. It behaves like an indentation key (which is expected in a code editor), but this results in an infinite tab trap, making it impossible for users to navigate to any UI components outside the editor using only the keyboard

![Image](https://github.com/user-attachments/assets/5763b7cf-b0c0-4b3a-aeaf-c06383b76e32)

### Expected Behavior

Users should be able to press a keyboard shortcut (e.g., Esc or Ctrl + M) to exit the editor and resume normal tab focus movement across the interface.

Alternatively, consider adding an invisible but focusable element after the editor to allow users to tab out.

### Steps:

1. Open the [p5.js Web Editor](https://editor.p5js.org/)
2. Click inside the code editor
3. Press the Tab key repeatedly
4. Notice that focus never leaves the editor

// Paste your code here :)

```

**Comment by raclim** [comment]
Thanks so much for opening this issue, this is a great catch! 

Just a quick note: since we're currently in the process of upgrading CodeMirror to version 6, it might be best to use v6 as the reference point for any further investigation into this issue. Alternatively, it could make sense for us to wait until the new version is fully released. I’m going to temporarily add the “On Hold” label to reflect that, but just something to keep in mind here!

**Comment by yugalkaushik** [comment]
While working on this issue I noticed many buttons and components in editor lacks tab behavior. Is it from some specific CSS or something.

**Comment by kitlord** [comment]
@raclim I found a quick fix for this issue, if we press shift+tab instead of tab while in the editor it will regain the focus but this solution may not be widely known for the users though. For the buttons which are not focused using tab we might need to open up a new issue instead.
